src/runtimes/node/bundlers/esbuild/bundler_target.ts

```diff
@@ -25,13 +24,12 @@ const getBundlerTarget = (suppliedVersion?: string): VersionValues => {
 
 const getModuleFormat = async (
   srcDir: string,
-  featureFlags: FeatureFlags,
   configVersion?: string,
 ): Promise<{ includedFiles: string[]; moduleFormat: ModuleFormat }> => {
   const packageJsonFile = await getClosestPackageJson(srcDir)
   const nodeSupport = getNodeSupportMatrix(configVersion)
 
-  if (featureFlags.zisi_pure_esm && packageJsonFile?.contents.type === 'module' && nodeSupport.esm) {
```

Comment:
I don't think we want to remove this one?

Reply:
Sorry, I know the PR is still in draft so I should've waited. It's just that this functionality caused me a huge headache in the past and the idea that this could be merged accidentally made me shiver! 😅 💦